### PR TITLE
Add kotlin code for the chapter of stack and queue

### PR DESCRIPTION
**Pull request by curtishd**
If this pull request (PR) pertains to **Chinese-to-English translation**, please confirm that you have read the contribution guidelines and complete the checklist below:

- [ ] This PR represents the translation of a single, complete document, or contains only bug fixes.
- [ ] The translation accurately conveys the original meaning and intent of the Chinese version. If deviations exist, I have provided explanatory comments to clarify the reasons.

If this pull request (PR) is associated with **coding or code transpilation**, please attach the relevant console outputs to the PR and complete the following checklist:

- [x] I have thoroughly reviewed the code, focusing on its formatting, comments, indentation, and file headers.
- [x] I have confirmed that the code execution outputs are consistent with those produced by the reference code (Python or Java).
- [x] The code is designed to be compatible on standard operating systems, including Windows, macOS, and Ubuntu.
屏幕截图 2024-03-19 113627
屏幕截图 2024-03-19 113640
屏幕截图 2024-03-19 113647
屏幕截图 2024-03-19 113656
屏幕截图 2024-03-19 113704
屏幕截图 2024-03-19 113712
屏幕截图 2024-03-19 113719
屏幕截图 2024-03-19 113726
屏幕截图 2024-03-19 113733

### CODE REVIEW

File: codes/kotlin/chapter_stack_and_queue/array_deque.kt [hunk 1]
```diff
@@ -0,0 +1,144 @@
+/**
+ * File: array_deque.kt
+ * Created Time: 2024-1-25
```
Comment (krahets) [inline codes/kotlin/chapter_stack_and_queue/array_deque.kt:3]: 2024-01-25

Comment (krahets) [reply codes/kotlin/chapter_stack_and_queue/array_deque.kt:3]: Please check all the code and unify them

**Review by krahets — APPROVED** [review]
Thanks for the PR! I will add Kotlin blocks to the documents

**Comment by curtishd** [comment]
You are welcome, it`s over with two chapters left.

**Pull request merged**